internal/acctest/authentication/access_token_test.go

```diff
@@ -0,0 +1,101 @@
+package auth_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/acctest"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/provider"
+)
+
+type Response struct {
+	AccessToken string `json:"access_token"`
+}
+
+func getAccessToken() {
+	oauthClientId := os.Getenv("PINGFEDERATE_PROVIDER_OAUTH_CLIENT_ID")
+	oauthClientSecret := os.Getenv("PINGFEDERATE_PROVIDER_OAUTH_CLIENT_SECRET")
+	os.Getenv("PINGFEDERATE_PROVIDER_OAUTH_TOKEN_URL")
+	os.Getenv("PINGFEDERATE_PROVIDER_OAUTH_SCOPES")
+	//#nosec G101
+	tokenRequestUrl := "https://localhost:9031/as/token.oauth2"
+
+	//#nosec G402
+	client := &http.Client{Transport: acctest.GetTransport()}
+	clientInfo := fmt.Sprintf("client_id=%s&grant_type=client_credentials&client_secret=%s&scope=email", oauthClientId, oauthClientSecret)
+	jsonBodyReader := strings.NewReader(clientInfo)
+	resp, err := client.Post(tokenRequestUrl, "application/x-www-form-urlencoded", jsonBodyReader)
+	if err != nil {
+		return
```

Comment:
This should propagate the error up so that there is useful output when it fails. Same for the other returns in this method